### DISCUSSION TITLE
Could list responses also be converted to JSON automatically?

### DISCUSSION THREAD

**Discussion by MaicoTimmerman** [pallets/flask#4533] [Ideas]
In https://github.com/pallets/flask/commit/7bf8366970ec49b5396a1f86f00fdd1c5fef3fed dictionaries returned from view function are automatically converted to JSON. Is there a reason why we do not do the same for lists? According to [RFC4627](https://www.rfc-editor.org/rfc/rfc4627) an array is a valid top level JSON construct.

I'd be open to contribute a change for this.

**Comment by davidism** [comment]
Because lists look too much like tuples, which already have a meaning.

There is a lot of discussion in the PR that is linked from the commit you linked. https://github.com/pallets/flask/pull/3111

**Reply by MaicoTimmerman** [reply]
Thanks for the link, I didn't see the PR in the commit.

Given the discussion regarding not forcing API design, I'd argue that current behavior of producing an error steers heavily towards returning objects, instead of lists.  I'd be +1 on adding `list` to the automatic serialization.

That said, I'll workaround using the following snippet:

```python
class TheFlask(Flask):
    def make_response(self, rv: Any):
        if isinstance(rv, list):
            rv = jsonify(rv)
        return super().make_response(rv)
```
